regression/pages/studio/studio_textbooks.py

```diff
@@ -0,0 +1,50 @@
+"""
+Extended Textbooks page
+"""
+from edxapp_acceptance.pages.studio.textbook_upload import TextbookUploadPage
+
+from regression.pages import UPLOAD_FILE_DIR
+from regression.tests.helpers import get_url
+
+
+class TextbookPageExtended(TextbookUploadPage):
+    """
+    Extended Page for textbook page
+    """
+    @property
+    def url(self):
+        """
+        Construct a URL to the page within the course.
+        """
+        url = get_url(self.url_path, self.course_info)
+        return url
+
+    def upload_textbook(self, file_name):
+        """
+        Uploads a pdf textbook
+        """
+        self.q(css='.action.action-upload').click()
+        self.q(css='.upload-dialog input').results[0].send_keys(
+            UPLOAD_FILE_DIR + "/" + file_name)
+        self.q(css='.button.action-primary.action-upload').click()
+        self.wait_for_element_absence(
+            ".modal-window-overlay", "Upload modal closed")
```

Comment:
@Muddasser wait_for_element_absense is doing the same. Steps are:
1- Click upload
2- It uploads and after the upload is completed the 'Upload Modal' closes automatically
I am using it as a verification. Makes sense?